src/odemis/driver/andorshrk.py

```diff
@@ -122,11 +147,11 @@ def at_errcheck(self, result, func, args):
         """
         # everything returns SHAMROCK_SUCCESS on correct usage
         if result not in ShamrockDLL.ok_code:
-            errmsg = create_string_buffer(ERRORLENGTH).decode('latin1')
+            errmsg = create_string_buffer(ERRORLENGTH)
             self.ShamrockGetFunctionReturnDescription(result, errmsg, len(errmsg))
             raise ShamrockError(result,
                                 "Call to %s failed with error %d: %s" %
-                                (func.__name__, result, errmsg.value))
+                                (func.__name__, result, to_str_escape(errmsg.value)))
```

Comment:
That looked better before. The error message should be a english message, so it shouldn't be escaped. I think it'd typically be ascii, but using latin1 is a safe bet.